packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -111,23 +113,27 @@ const ProductDetail = ({category, product, isLoading}) => {
             status: 'error'
         })
     }
-    const handleAddToCart = async (variant, quantity) => {
+
+    const handleAddToCart = async (productSelectionValues) => {
+        let returnVal
+
         try {
-            if (!variant?.orderable || !quantity) return
-            // The basket accepts an array of `ProductItems`, so lets create a single
-            // item array to add to the basket.
-            const productItems = [
-                {
-                    productId: variant.productId,
-                    quantity,
-                    price: variant.price
-                }
-            ]
+            const productItems = productSelectionValues.map(({variant, quantity}) => ({
+                productId: variant.productId,
+                price: variant.price,
+                quantity
+            }))
 
             await basket.addItemToBasket(productItems)
+
+            // If the items were sucessfully added, set the return value to be used
+            // by the add to cart modal.
+            returnVal = productSelectionValues
```

Comment:
```suggestion
            return productSelectionValues
```
If you're not doing anything with `returnVal`, might as well just return...